File: tests/Staford.NLP.Tools/Java.fs
module Stanford.NLP.Tools.Java

open java.lang
open java.util

let toSeq (iter:Iterable) =
    let rec loop (x:Iterator) =
        seq {
            if x.hasNext() then
                yield x.next()
                yield! (loop x)
        }
    iter.iterator() |> loop |> Array.ofSeq |> Seq.readonly

let props (values) =
    let props = Properties()
    for (key, value) in values do
        props.setProperty(key, value) |> ignore
    props

File: tests/Stanford.NLP.Tools/Java.cs
using System.Collections.Generic;
using IKVM.Runtime;
using java.lang;
using java.util;

namespace Stanford.NLP.Tools
{
    public static class Java
    {
        public static IEnumerable<object> ToSeq(this Iterable iterable)
        {
            var iterator = iterable.iterator();
            while (iterator.hasNext())
            {
                yield return iterator.next();
            }
        }

        public static Properties Props(Dictionary<string, string> values)
        {
            var props = new Properties();
            foreach (var kv in values)
            {
                props.setProperty(kv.Key, kv.Value);
            }
            return props;
        }
    }
}
